### ISSUE TITLE
When using Test.createTestingModule(), Change to Injector.ts is Swallowing Exceptions with a Maximum Call Stack Error

### ISSUE THREAD

**Issue by camoore1979** [nestjs/nest#14347]
### Did you read the migration guide?

- [X] I have read the whole migration guide

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Potential Commit/PR that introduced the regression

PR #14133 / Commit 229d97f01837e8cf1d50cda57ac6cfe1c0235bc3

### NestJS version

10.1.2 -> 10.4.15

### Describe the regression

We have a fairly large and complex Nest.js application and in a recent upgrade from 10.1.2 to 10.4.15, our application would run successfully but integration test suites began failing with a `RangeError: Maximum call stack size exceeded` exception when attempting to setup our App module for testing with `Test.createTestingModule`. We were getting the error in almost any case even when only creating a testing module with as few modules as possible. Eventually we realized that exceptions thrown in our code were being caught and swallowed. For instance we could leave out a required environment variable from our Config service and it was throwing the expected error for the undefined configuration value but this was masked by the `RangeError`.  Eventually, we determined that this was caused by recent changes to `packages/core/injector/injector.ts`, we were able to comment out the changes locally and our test would throw the expected exception.

Thus, in fairly complex Nest.js Apps this recent change may be swallowing exceptions and throwing a misleading exception.

### Input code

```ts

```

### Expected behavior

Exceptions in application code should not be swallowed by catches within the Nest.js framework. 

### Other

Node.js 22.11.0, Typescript 5.7.2

![image](https://github.com/user-attachments/assets/ffddcfb7-5900-488e-a564-847732d929ab)

**Comment by kamilmysliwiec** [comment]
Could you please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project)? This would allow us to fix this issue much faster

**Comment by fgodino-milu** [comment]
@camoore1979 were you able to fix this?

**Comment by camoore1979** [comment]
No. We backed off versions to 10.4.5 just prior to the change which introduced the problem. At some we'll have to fix the issue within our code (our module deps) or need a fix applied to @nestjs/core by the community (or try to contribute one).
